### PR TITLE
v2.6.0.0: Port to Selenium-only, PHP >= 8.1, removed Discover & Trending and more

### PR DESCRIPTION
**Pull request by pablouser1**
**TL;DR**: TikScraper now enforces Selenium, if you have PHP < 8.1 you should update and the discover and trending methods have been removed.
Also make sure to check the new config array found in `README.md`, it has been changed significantly.

👋 Hi, long time no see!

As you may already be aware, the library has been broken for quite some time. This is because TikTok has been adding some anti-scraping methods that, with the old library, were very annoying to bypass.

The old signing method was very lightweight but was hard to keep updated, specially when the new signing process requires to use DOM APIs not implemented (or not fully implemented) in JSDOM.

Having to navigate through obfuscated code was annoying enough, so I've finally decided to enforce Selenium / Chromedriver usage. I'll detail a little bit, for those curious, how the old system worked and how the new one does.

### Request system
The old system was made from:
* the library itself, which sent the requests using Guzzle
* the signing service, where the URL GET params gets sent to a signer (usually SignTok, in case of this library) that ran TikTok's obfuscated code and signed the URL.

These two systems barely had data in common (cookies, user-agent...) making being detected pretty easy.

Now, the new system is made from 2ish components:
* the library itself, Guzzle is only used for fetching videos, it shares cookies and user agent with selenium
* selenium, used for sending requests to TikTok

Working with a selenium instance makes detection much harder as they are essentially browser.

It is still vulnerable to IP blacklist, but there's not much you can do against that, other than using a proxy.

### Tokens
TikScraper now accepts custom `verifyFp` and `device_id` tokens. This is specially useful for bypassing captcha requests and `Forbidden` errors when streaming videos.

The main issue with the verifyFp token (and a BIG problem for the project itself) is that it expires, and the only way to get this token (at least for now) is to solve a captcha. You usally get a captcha when:
- You make too many requests
- You use a non-residential IP

### PHP
The library now requires PHP >= 8.1, most Linux distributions already have at least that version on their stable repos, so upgrading it shouldn't be an issue.

### Docs
I'm working on implementing a Github Pages site with documentation for the library made using phpdocumentator

### Removed features
- Discover, this method used TikTok's older endpoints `/node/...`. Although it's still working, the new `/api/explore` endpoint is available and will be implemented in the future.
- Trending, this method used the endpoint `/api/recommend/item_list`, as far as I'm aware, this is the endpoint for the "For you" page. Having a for you page for shared instances doesn't make much sense so I decided to remove it.

### To be implemented
With the new system, implementing new features will be easier, the following methods will be implemented in the v2.6.x.x version:
- Search
- Comments
- Live (maybe)

### CODE REVIEW

**Comment by manoflinux** [comment]
I disagree with your decision to get rid of the trending tab, this was the only way I could get a rss feed of trending videos. 
I tried the previous version but I get an error when I try the trending button. 
Could please help me fix the previous version and once have something working I can make a fork of it, I might  even strip out all the web interface stuff because all I need is the rss feed.

**Comment by manoflinux** [comment]
going a bug report on right now.
